java/src/jmri/jmrit/operations/rollingstock/RollingStock.java

```diff
@@ -849,17 +849,8 @@ public void setRfid(String id) {
             log.debug("Setting IdTag for {} to {}", this, id);
             _rfid = id;
             if (!id.equals(NONE)) {
-                try {
-                    IdTag tag = InstanceManager.getDefault(IdTagManager.class).getIdTag(id);
-                    if (tag != null) {
-                        log.debug("Tag {} found", tag);
-                        setIdTag(tag);
-                    } else {
-                        log.error("Tag {} not found", id);
-                    }
-                } catch (NullPointerException e) {
-                    log.error("Tag manager not found");
-                }
+                IdTag tag = InstanceManager.getDefault(IdTagManager.class).provideIdTag(id);
+                setIdTag(tag);
```

Comment:
I'd be tempted to check that an IDTag is actually provided when _potentially_ creating any NamedBean,
 eg.
```
                try {
                    setIdTag(InstanceManager.getDefault(IdTagManager.class).provideIdTag(id));
                } catch (IllegalArgumentException e) {
                    log.error("Unable to provide IDTag \"{}\" {}",id, e.getMessage());
                }
```
If the code is in a loop, then at least it won't crash out mid-run, just notify on any failing ones.

With the InternalIdTagManager it's pretty difficult to create an exception with the code as is proposed,
however no harm in adding the extra bit of protection.

Steve.